src/App.tsx

```diff
@@ -1,27 +1,72 @@
-import React from 'react';
+import React, { useCallback, useState } from 'react';
 import './App.scss';
 import { GoodsList } from './GoodsList';
 
-// import { getAll, get5First, getRed } from './api/goods';
+import { getAll, get5First, getRedGoods } from './api/goods';
+import { Good } from './types/Good';
 // or
 // import * as goodsAPI from './api/goods';
 
-export const App: React.FC = () => (
-  <div className="App">
-    <h1>Dynamic list of Goods</h1>
+export const App: React.FC = () => {
+  const [goods, setGoods] = useState<Good[]>([]);
 
-    <button type="button" data-cy="all-button">
-      Load all goods
-    </button>
+  const handleClick = useCallback((buttonName: string) => {
+    switch (buttonName) {
+      case 'all-button':
+        getAll()
+          .then(apiGoods => setGoods(apiGoods));
```

Comment:
```suggestion
        getAll()
          .then(setGoods);
```
We can shorten the code